src/wlapplication.cc

```diff
@@ -211,6 +211,31 @@ bool is_autogenerated_and_expired(const std::string& filename,
 	return std::difftime(time(nullptr), mktime(&tfile)) > keep_time;
 }
 
+// A simple wrapper around the game logic thread. It ensures the thread is
+// always joined when this class goes out of scope. This is important when an
+// exception is thrown within the game anywhere, because without this, we will
+// not join the thread and this means the exception will never be caught and
+// printed, making debugging hard.
+// See
+// https://stackoverflow.com/questions/25397874/deleting-stdthread-pointer-raises-exception-libcabi-dylib-terminating
+class GameLogicThread {
```

Comment:
I think this class will be useful for other purposed as well. How about:
- Move it to base/multithreading.h
- Rename it to something like ThreadWrapper
- Modify the constructor to accept any function, not just the default one
- Make `should_die_` optional so it can be `nullptr` for temporarily used threads

Reply:
2c: [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it), and if I am wrong about this, pull it out at that point. For now, pulling this helper class into a higher scope means that it is more public, and hence harder to change or remove in the future again. Trying to keep code on the most local and private level as possible usually leads to less coupled and easier to change code base. I do not have strong feelings in this specific case.

If you pull it on a higher level, let's not take a pointer to `should_die_`, instead let's take a `std::function` that is called in the place that currently sets `should_die_`. @klaus-halfmann fyi too.